File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int localSum = nums[0];
        int globalSum = nums[0];
        for (int i = 1; i < nums.length; i++) {
            int num = nums[i];
            localSum = Math.max(num, localSum+num);
            globalSum = Math.max(localSum, globalSum);
        }
        return globalSum;  
    }
}

File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        global_sum = local_sum = float('-inf')
        
        for num in nums:
            local_sum = max(local_sum + num, num)
            global_sum = max(local_sum, global_sum)
        
        return global_sum
        
        
        
        